app/src/main/java/com/example/dispatcher/presentation/homepage/viewModel/ArticlesViewModel.kt

```diff
@@ -25,18 +27,16 @@ class ArticlesViewModel(application: Application) : AndroidViewModel(application
     }
 
     private fun fetchArticles() {
-        articlesRepository.fetchArticles().observeForever { articleUiModels ->
-            _articlesLiveData.postValue(articleUiModels)
+        viewModelScope.launch {
```

Comment:
This way, you run the API request in the Main (UI) thread. we want to run the api in the background thread so the ui won't affect by it. 
 what inside ` viewModelScope.launch` is performed by the UI thread.

so you need to wrap the call: 
`viewModelScope.launch {
val topHeadlines = withContext(Dispatchers.IO) {
articlesRepository.fetchArticles()
}
_articlesLiveData.postValue(topHeadlines)
}
`
this way -  the call is running by bg thread,and the livedata update is by the UI thread.

please make sure you apply this comment in all other APIs calls